DataAccess/Entity/Entities/PO.cs

```diff
@@ -18,8 +18,8 @@ public class PO : BaseEntity
         public float NetAmount { get; set; }
         [Required]
         [Index]
-        public DateTime TransactionTime { get; set; }
-        public int EmployeeID { get; set; }
+        public DateTime TransactionTime { get; set; } = DateTime.UtcNow;
+        public int EmployeeID { get; set; } = 1;

```

Comment:
I iz lazy

Reply:
Having a default datetime is fine. It's not a good idea to default the employeeID, because an employee record with id=1 might not exist. I've hard coded employeeID=1 in the code for now because I don't have an employee chooser yet.

Reply:
It's the effective date of the order. Not necessarily the same as the time it was created. What I was planning to do, was to put the fields   CreatedBy, CreatedTime, UpdatedBy, UpdateTime  on basically all the tables in the database. Maybe I'll have them populated more automatically in some base entity class. It will use the UserID currently logged into the program. Both of the retail software places I worked had this in nearly all tables.

Reply:
That looks pretty interesting. But not really needed for the HOApp, because only a small number of users might be accessing the same DB at once, with maybe 10 instances of the program. Recording an audit trail of database transactional changes would be necessary for a data replication system. My data replication right now is very simple and one-way. In a real system it would use a queue of replication tasks (transactional DB changes).

Reply:
heh i didn't mean for this app, was just something i've been reading up on, thought i'd share 😀

Reply:
Incidentally event sourcing isn't an audit log, it's the foundation. There's no relational db backing it up. Models are generated from replaying events basically. It generally goes hand in hand with CQRS: https://msdn.microsoft.com/en-us/library/jj591573.aspx